src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+
+    private static final DateTimeFormatter DATES = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public LocalDate parseDate(String date) {
+        try {
+            return LocalDate.parse(date, DATES);
+        } catch (Exception e) {
+            throw new InvalidDataException(
+                    "Invalid date format. Only dd.MM.yyyy format is accepted"
+            );
+        }
+    }
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        if (names == null || data == null || dateFrom == null || dateTo == null) {
+            throw new InvalidDataException("Invalid data: Please check your input");
+        }
+
+        LocalDate dateFromFormatted = parseDate(dateFrom);
+        LocalDate dateToFormatted = parseDate(dateTo);
+
+        StringBuilder output = new StringBuilder("Report for period "
+                + DATES.format(dateFromFormatted)
+                + " - "
+                + DATES.format(dateToFormatted));
+
+        for (String name : names) {
+            int salary = 0;
+            for (String part : data) {
+                String[] dataPart = part.split(" ");
+                LocalDate datePart = LocalDate.parse(dataPart[0], DATES);
```

Comment:
Consider using the `parseDate` method here instead of directly parsing the date. This will ensure consistent date parsing and error handling.